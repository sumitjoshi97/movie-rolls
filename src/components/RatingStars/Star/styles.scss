.star {
  display: inline-block;
  &__icon {
    width: 1.7rem;
    height: 1.7rem;
    fill: #ff0000;
  }
}
