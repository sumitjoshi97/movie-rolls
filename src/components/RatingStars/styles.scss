.rating-stars {
  display: flex;
  padding-left: 1rem;
}
