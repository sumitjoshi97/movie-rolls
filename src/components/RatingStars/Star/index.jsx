import React, { Component } from 'react'
import MdStar from 'react-ionicons/lib/MdStar'
import MdStarOutline from 'react-ionicons/lib/MdStarOutline'
import MdStarHalf from 'react-ionicons/lib/MdStarHalf'
import PropTypes from 'prop-types'

import './styles.scss'

export default class Star extends Component {
  static propTypes = {
    type: PropTypes.string.isRequired
  }

  renderStar = starType => {
    switch (starType) {
      case 'half':
        return <MdStarHalf {...options} />
      case 'blank':
        return <MdStarOutline {...options} />
      default:
        return <MdStar {...options} />
    }
  }

  render() {
    return this.renderStar(this.props.type)
  }
}

const options = {
  color: '#ff0000'
}
