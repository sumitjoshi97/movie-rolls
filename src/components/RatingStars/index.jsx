import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Star from './Star'

import './styles.scss'

export default class RatingStars extends Component {
  static propTypes = {
    stars: PropTypes.number
  }

  renderStars = () => {
    let { stars } = this.props
    stars = stars / 2
    let renderStars = []

    for (let i = 1; i <= 5; i++) {
      if (i > stars && i === Math.ceil(stars)) {
        renderStars.push(<Star key={i} type='half' />)
      } else if (i <= stars) {
        renderStars.push(<Star key={i} type='full' />)
      } else {
        renderStars.push(<Star key={i} type='blank' />)
      }
    }
    return renderStars
  }
  render() {
    return <div className='rating-stars'>{this.renderStars()}</div>
  }
}
