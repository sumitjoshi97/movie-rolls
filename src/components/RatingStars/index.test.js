import React from 'react'
import { shallow } from 'enzyme'

import RatingStars from './index'
import Star from './Star'

describe('rating star component unit test', () => {
  const stars = 5

  const props = { stars }

  const ratingStarsWrapper = shallow(<RatingStars {...props} />)

  test('should render properly', () => {
    expect(ratingStarsWrapper).toMatchSnapshot()
  })

  test('should render `5 stars`', () => {
    expect(ratingStarsWrapper.find(Star)).toHaveLength(5)
  })

  test('should render full `2 stars`', () => {
    expect(ratingStarsWrapper.find({ type: 'full' }).find(Star)).toHaveLength(2)
  })
  test('should render a `half star`', () => {
    expect(ratingStarsWrapper.find({ type: 'half' }).find(Star)).toHaveLength(1)
  })

  test('should render empty `2 stars`', () => {
    expect(ratingStarsWrapper.find({ type: 'blank' }).find(Star)).toHaveLength(
      2
    )
  })
})
