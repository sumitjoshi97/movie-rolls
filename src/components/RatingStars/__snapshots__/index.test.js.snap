// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rating star component unit test should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RatingStars
    stars={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Star
          type="full"
        />,
        <Star
          type="full"
        />,
        <Star
          type="half"
        />,
        <Star
          type="blank"
        />,
        <Star
          type="blank"
        />,
      ],
      "className": "rating-stars",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "type": "full",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "type": "full",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "type": "half",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "type": "blank",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "class",
        "props": Object {
          "type": "blank",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Star
            type="full"
          />,
          <Star
            type="full"
          />,
          <Star
            type="half"
          />,
          <Star
            type="blank"
          />,
          <Star
            type="blank"
          />,
        ],
        "className": "rating-stars",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "type": "full",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "type": "full",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "type": "half",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "type": "blank",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "class",
          "props": Object {
            "type": "blank",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
